pkg/apis/nodeup/config.go

```diff
@@ -64,6 +64,8 @@ type Config struct {
 	KubeProxy *kops.KubeProxyConfig
 	// Networking configures networking.
 	Networking kops.NetworkingSpec
+	// API controls how the Kubernetes API is exposed.
+	API *kops.APISpec `json:",omitempty"`
```

Comment:
We only copy fields that affect the configurations of nodes of the appropriate type. Changes to NodeupConfig result in nodes of the relevant instance group being selected for rolling update, whereas changes to the ClusterSpec do not. So since we only populate this new field for control-plane and api-server nodes, changes to the sub-fields we populate result in control-plane and api-server nodes being selected for update, but worker nodes not. Previously, no nodes would be selected for update, so control-plane and api-server nodes would continue to run with outdated configuration.